ref: Handle downloading IdP xml with sentry.http

evanpurkhiser opened the pull request:
Keeping http fetches on the same codepath, helping to avoid security concerns.

evanpurkhiser committed:
ref: Handle downloading IdP xml with sentry.http

mattrobenolt commented on src/sentry_auth_saml2/forms.py:
```diff
@@ -23,7 +25,8 @@ def extract_idp_data_from_parsed_data(data):
 def process_url(form):
     url = form.cleaned_data['metadata_url']
-    data = OneLogin_Saml2_IdPMetadataParser.parse_remote(url)
+    url_result = fetch_file(url)
```
Don't use `fetch_file`. That might be slightly misleading, but this has extra behavior that we don't need or want compared to just `safe_urlopen`.

`fetch_file` adds caching and stuff for event processing pipeline which we don't want here.

evanpurkhiser replied:
Ahhh, sounds good, should have looked closer.

evanpurkhiser replied:
https://github.com/getsentry/sentry-auth-saml2/pull/19/commits/088259fff08abbcf766963a9ddfbfce7b0a6b9f6

evanpurkhiser commented:
yep, just fixing that

evanpurkhiser commented:
Going to split this into two commits actually, one for sentry.http the other for the encoding fix, same PR though then will rebase merge.

evanpurkhiser committed:
fix: Encode form string as utf8

This fixes an issue where lxml throws it's hands in the air if it
receives a unicode string with a xml declaration [1].

[1]: https://github.com/lxml/lxml/blob/904c9742b23914ce772a4c10013855b0996d5f73/src/lxml/parser.pxi#L1863-L1868
